bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -234,10 +234,10 @@ public Rule add(Rule rule) {
         String rUID = rule.getUID();
         if (rUID == null) {
             rUID = ruleEngine.getUniqueId();
-            super.add(initRuleId(rUID, rule));
-        } else {
-            super.add(rule);
+            rule = initRuleId(rUID, rule);
         }
+        ruleEngine.normalizeRuleConfigurations(rule);
```

Comment:
@sjka How do we do config normalisation on the Thing API? Don't we already expect it to be normalised when being added to the registry? We should imho make sure that we are using the same principals in both situations.

Reply:
The thing that worries me here is that I would expect the rule engine to treat the passed Rule object as immutable and not to do any changes to it. If we are concerned that we might see configurations that are not normalized (and we cannot deal with them), we could throw IAEs to make sure that only expected stuff is reached in. Nonetheless, I also see the point that the framework could do this work if it anyhow knows what needs to be done. 
@sjka & @maggu2810  What do you think how we should deal with that?

Reply:
> My point of view is that ThingRegistry is a public service which has a public method updateConfiguration(). There is no guarantee that every one will call the method with normalize parameters.

￼Sure, no guarantee, but the public method could state it requires a normalized configuration.

> we could throw IAEs to make sure that only expected stuff is reached in

Okay for me.

Reply:
> With Things, we indeed try to normalize them as early as possible

It is the same for Rules (there is a bug in `ConfigurationDeserializer`, but I'll make a separate PR for it). Additionally, in Rule Registry we also need normalization, because the rule configuration is resolved with default values and references. After the resolving, normalization is needed because the default values are always strings. This however will be fixed with #4468, so this PR can be closed without merge.